lib/rake/application.rb

```diff
@@ -155,19 +155,24 @@ def thread_pool             # :nodoc:
 
     # Invokes a task with arguments that are extracted from +task_string+
     def invoke_task(task_string) # :nodoc:
-      name, args = parse_task_string(task_string)
+      name, args, safe_call = parse_task_string(task_string)
+      return if safe_call && !lookup(name)
+
       t = self[name]
       t.invoke(*args)
     end
 
     def parse_task_string(string) # :nodoc:
-      /^([^\[]+)(?:\[(.*)\])$/ =~ string.to_s
+      string = string.to_s.dup
+      safe_call = !!string.chomp!('?')
+
+      /^([^\[]+)(?:\[(.*)\])$/ =~ string
 
       name           = $1
       remaining_args = $2
```

Comment:
```suggestion
      /^([^\[]+)(?:\[(.*)\])(\?)?$/ =~ string.to_s

      name           = $1
      remaining_args = $2
      safe_call      = !!$~.begin(3)
```

Reply:
@nobu I tried that but it doesn't work for various reasons. Mostly because simple tasks without arguments actually don't match that pattern.